spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/keras/Bidirectional.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.nn.keras
+
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.nn.abstractnn.AbstractModule
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.{Shape, Table}
+
+import scala.reflect.ClassTag
+
+class Bidirectional[T: ClassTag](
+   val layer: Recurrent[T],
+   val mergeMode: String = "concat",
+   var inputShape: Shape = null)(implicit ev: TensorNumeric[T])
+  extends KerasLayer[Tensor[T], Tensor[T], T](KerasLayer.addBatch(inputShape)) {
+
+  private val mode = mergeMode.toLowerCase()
+
+  require(layer.returnSequences, "Bidirectional requires RNNs to return the full sequence")
+  require(mode == "sum" || mode == "mul" || mode == "concat" || mode == "ave",
+    s"Invalid merge mode: $mergeMode")
+
+  override def computeOutputShape(inputShape: Shape): Shape = {
+    val output = layer.computeOutputShape(inputShape)
+    if (mode == "concat") {
+      val outputArray = output.toSingle().toArray
+      outputArray(outputArray.length-1) = outputArray(outputArray.length-1) *2
```

Comment:
outputArray.last

Reply:
If I change to outputArray.last, it will report error `can't assignment to val` even if I change outputArray to var....